sys/phydat/phydat_str.c

```diff
@@ -78,14 +80,35 @@ const char *phydat_unit_to_str(uint8_t unit)
     }
 }
 
-char phydat_scale_to_str(int8_t scale)
+char phydat_scale_to_str(int8_t scale, uint8_t unit)
 {
+    /* extra handling for multi and non dimensional units */
+    switch (unit) {
+        case UNIT_M2:       /* rescale for 2d unit */
+            if (scale % 2 != 0) {
```

Comment:
Yes, and that would be still possible. But output with this function would be broken (which it is still in your implementation, since you decided to just return an empty string, which might be confusing). A user that is using this function should know (via the doc) that using the function with a scale not divisible by 2 or 3 for the respective unit, so we can just `assert()` to help with debugging.